internal/controller/management_controller.go

```diff
@@ -142,6 +158,54 @@ func (r *ManagementReconciler) Update(ctx context.Context, management *hmc.Manag
 	return ctrl.Result{}, nil
 }
 
+// checkProviderStatus checks the status of a provider associated with a given
+// ProviderTemplate name.  Since there's no way to determine resource Kind from
```

Comment:
Is there a way we could add `ProviderTypes` to our templates (providers like `k0smotron` are both infra and bootstrap providers) or extract the `Kind`'s from the `provider.yaml`'s somehow?  If so, we don't need to do this weird iteration thing.